catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -415,17 +415,24 @@ public CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
     Stopwatch stopwatch = Stopwatch.createStarted();
 
     List<QueryResponse> responses = Collections.synchronizedList(new ArrayList<>());
-    QueryFunction queryFunction =
-        (queryRequest) -> {
-          QueryResponse queryResponse = catalogFramework.query(queryRequest);
-          responses.add(queryResponse);
-          return queryResponse;
-        };
-
-    List<Result> results =
-        ResultIterable.resultIterable(queryFunction, request, cqlRequest.getCount())
-            .stream()
-            .collect(Collectors.toList());
+
+    List<Result> results;
+    if (cqlRequest.getCount() == 0) {
+      QueryResponse queryResponse = catalogFramework.query(request);
+      results = queryResponse.getResults();
+      responses.add(queryResponse);
+    } else {
+      QueryFunction queryFunction =
+          (queryRequest) -> {
+            QueryResponse queryResponse = catalogFramework.query(queryRequest);
+            responses.add(queryResponse);
+            return queryResponse;
+          };
+      results =
+          ResultIterable.resultIterable(queryFunction, request, cqlRequest.getCount())
+              .stream()
+              .collect(Collectors.toList());
```

Comment:
❓Are we putting a limit on the `cqlRequest.getCount()` to make sure we don't blow up the heap when we collect all the results?

Reply:
☝️
https://github.com/codice/ddf/pull/3769#discussion_r218979703

Reply:
I think the Catalog Framework protects against this already. 
https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java#L136

Reply:
Since this problem existed before this PR; we shouldn't require a resolution for approval